src/main/java/ttt/players/DelayedPlayer.java

```diff
@@ -7,32 +7,47 @@
 public class DelayedPlayer implements Player {
 
     private Player player;
+    private Integer delay;
 
-    public DelayedPlayer(Player player) {
+    public DelayedPlayer(RandomPlayer player) {
```

Comment:
Referring to different concrete types here starts to violate OCP. (If a new type of player came along, you'd need to add another constructor here).

Instead, could the delay be passed into the constructor, and the PlayerFactory decide whether a Delayed Player is needed, and if so, pass in the relevant timeout? This way all the Player type checking would be in one place.